presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -446,6 +449,16 @@ public static NullableValue parsePartitionValue(String partitionName, String val
             return NullableValue.of(TIMESTAMP, timestampPartitionKey(value, timeZone, partitionName));
         }
 
+        if (REAL.equals(type)) {
+            if (isNull) {
+                return NullableValue.asNull(REAL);
+            }
+            if (value.isEmpty()) {
+                return NullableValue.of(REAL, floatToRawIntBits(0.0f));
```

Comment:
I think this needs to be `return NullableValue.of(REAL, (long) floatToRawIntBits(0.0f));`, or you will end up with an `Integer` in `NullableValue` instead of a `Long`